### DISCUSSION TITLE
Animated arc layer - increasing arc segments, need advice

### DISCUSSION THREAD

**Discussion by gallickgunner** [visgl/deck.gl#5567] [Q&A] [answered]
![ezgif com-gif-maker](https://user-images.githubusercontent.com/40718428/110423346-b1cdd700-80c2-11eb-885c-56ec9daef311.gif)

Hello, I created an animation as shown in the gif above. I did this by extruding a specific arc segment found by the `segmentRatio` variable in the shader over time. The problem is the pings are quite long and I want them to be more ball like. I think this is due to the low number of segments. Also the elongated pings are more visibile at higher zoom levels since then a larger part of the screen would be rendering that specific segment.

1) Is there any way I can change the number of segments or perhaps decrease the length of the ping without increasing the segments and rendering overhead?

2) Secondly, I need a unique ID for every arc. Not arc layer but every arc in the arc layer so that I can randomize the starting times of the animation. Currently they are in sync. I'm currently using the `instancedSourcePosition` for this thing but if there are multiple arcs  from same source to target with a tilt parameter to distinguish them, the pings would again go in sync. I could use the `instanceTilts` attribute but is there any other sophisticated variable for this purpose? I'm so used to compute shaders, I was thinking of invocationIDs but I don't think vertex shaders have those if i'm right.

Thanks.

**Comment by Pessimistress** [comment, marked as answer]
You will need to subclass the ArcLayer. See https://deck.gl/docs/developer-guide/custom-layers/subclassed-layers

- override [_getModel](https://github.com/visgl/deck.gl/blob/master/modules/layers/src/arc-layer/arc-layer.js#L151) to use a custom resolution
- add your own attribute to the shader. [example](https://github.com/visgl/deck.gl/blob/master/examples/website/globe/animated-arc-layer.js)

**Reply by gallickgunner** [reply]
Ok, I did what you said and got it to work like i wanted. Thanks again pes.

**Reply by mikkurogue** [reply]
Hey @gallickgunner could you provide the example for the way you implemented the animation for the ArcLayer? I cant figure it out
